test/unitary/IQueryable.test.ts

```diff
@@ -2004,4 +2009,35 @@ export namespace IQueryableUnitTest
             Test.isArrayEqual(grouped.single(g => g.key === 3).value.toArray(), ["Ivan", "Uxue"]);
         });
     }
+
+    function zip(instancer: Instancer): void
+    {
+        it("Empty zip if empty sources", () =>
+        {
+            const numbers = [1, 2, 3];
+            const selector = (x: number, y: number) => x + y;
+
+            Test.isArrayEqual(instancer<number>([]).zip(<number[]>[], selector).toArray(), []);
+            Test.isArrayEqual(instancer<number>([]).zip(numbers, selector).toArray(), []);
+            Test.isArrayEqual(instancer<number>(numbers).zip(<number[]>[], selector).toArray(), []);
+        });
+
+        it("Simple zipping", () =>
+        {
+            const numbers = [1, 2, 3];
+            const letters = ["one", "two", "three"];
+
+            Test.isArrayEqual(instancer<number>(numbers).zip(letters, (x, y) => `${x}: ${y}`).toArray(), ["1: one", "2: two", "3: three"]);
+            Test.isArrayEqual(instancer<string>(letters).zip(numbers, (x, y) => x.length + y).toArray(), [4, 5, 8]);
+        });
+
+        it("Zipping of collections with unequal number of elements", () =>
+        {
+            const odd = [1, 3, 5, 7, 9];
+            const even = [2, 4, 6];
+
+            Test.isArrayEqual(instancer<number>(odd).zip(even, (x, y) => y - x).toArray(), [1, 1, 1]);
+            Test.isArrayEqual(instancer<number>(even).zip(odd, (x, y) => x * y).toArray(), [2, 12, 30]);
+        });
```

Comment:
Test proper deferred execution by
1. creating 2 empty lists
2. creating the Zip enumerable `list1.Zip(list2, ..)`
3. populate the lists
4. toArray() + assertion